cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py

```diff
@@ -64,6 +64,8 @@ def parse_bcl2fastq_raw_tile_metrics(
     )
 
     for stats_json_path in stats_json_paths:
+        if not stats_json_path.exists():
+            raise FileNotFoundError(f"File {stats_json_path} does not exist.")
```

Comment:
Nice! Do we want to update the doc string to specify which errors can be raised by the function? 

I think this would be a good addition to our coding conventions, to provide some specification on when and how we should write our doc strings. There are some existing standards, such as the Google Python standard https://google.github.io/styleguide/pyguide.html#383-functions-and-methods and PEP8 https://peps.python.org/pep-0257/ which we could apply.

Reply:
This is an example of the Google standard:
```
def fetch_smalltable_rows(
    table_handle: smalltable.Table,
    keys: Sequence[bytes | str],
    require_all_keys: bool = False,
) -> Mapping[bytes, tuple[str, ...]]:
    """Fetches rows from a Smalltable.

    Retrieves rows pertaining to the given keys from the Table instance
    represented by table_handle.  String keys will be UTF-8 encoded.

    Args:
        table_handle: An open smalltable.Table instance.
        keys: A sequence of strings representing the key of each table
          row to fetch.  String keys will be UTF-8 encoded.
        require_all_keys: If True only rows with values set for all keys will be
          returned.

    Returns:
        A dict mapping keys to the corresponding table row data
        fetched. Each row is represented as a tuple of strings. For
        example:

        {b'Serak': ('Rigel VII', 'Preparer'),
         b'Zim': ('Irk', 'Invader'),
         b'Lrrr': ('Omicron Persei 8', 'Emperor')}

        Returned keys are always bytes.  If a key from the keys argument is
        missing from the dictionary, then that row was not found in the
        table (and require_all_keys must have been False).

    Raises:
        IOError: An error occurred accessing the smalltable.
    """
```

Reply:
We can raise this in the next morning meeting. But as far is I gathered one-line doc strings were preferred previously.